types/setting.go

```diff
@@ -29,6 +29,12 @@ const (
 	ValueUnknown = "unknown"
 )
 
+const (
+	SettingUpdateFromAPI        = "api"
+	SettingUpdateFromController = "controller"
+	SettingUpdateFromUpgrade    = "upgrade"
+)
```

Comment:
Probably no need, if we set the empty value to the annotation.

Reply:
Removed. Thanks.